department-of-veterans-affairs/vets-website
# Certificate of Eligibility | Add unit tests for COE form app

## Original Ticket

[#46032](https://github.com/department-of-veterans-affairs/va.gov-team/issues/46032)

## URL of change

`/housing-assistance/home-loans/request-coe-form-26-1880/`

## Background

The COE form app is getting closer to release, we need to get code coverage up to around 80% (or more) - it is currently around [26.6% coverage for the form](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/frontend-support-dashboard/unit-test-coverage-report)

## Steps to test

1. Continuous integration tests are run in this PR - check that all are passing
2. Review unit test code

## Code changes

While writing unit tests for every component, minor changes were made:
- To allow for easier testing
- Removed unused code (e.g. `utils`)
- Removed duplicate code (e.g. `formatDate` replaced with platform `formDateLong`)
- Switched to use web components 
- General code cleanup
- Updated platform `FormStartControls` to prevent JS errors thrown while testing

## How was your code tested

Lots of unit tests!

## Acceptance criteria
- [x] Code cleanup & improvement

## Definition of done
